pkg/controller/hostpathprovisioner/daemonset.go

```diff
@@ -232,7 +232,7 @@ func (r *ReconcileHostPathProvisioner) getDuplicateDaemonSet(customCrName, names
 	}
 
 	for _, ds := range dsList.Items {
-		if ds.Name != MultiPurposeHostPathProvisionerName {
+		if ds.Name != MultiPurposeHostPathProvisionerName && ds.OwnerReferences[0].Kind == "HostPathProvisioner" {
```

Comment:
We should have the CR name here, so match on that as well.

Reply:
Done